cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/kubernetes/PvcOperator.java

```diff
@@ -102,4 +104,23 @@ protected void revertImmutableChanges(PersistentVolumeClaim current, PersistentV
         desired.getSpec().setSelector(current.getSpec().getSelector());
         desired.getSpec().setDataSource(current.getSpec().getDataSource());
     }
+
+    /**
+     * Method for configuring the normalized volume size to "desired" PVC from the "current" one.
+     * Used in situations when f.e. "1024Gi" is configured in the "desired" PVC, however Kubernetes during creation/update
+     * of PVC normalize it to "1Ti". In the {@link io.strimzi.operator.common.operator.resource.ResourceDiff} we assume that
+     * the values are different (because the String values are different), and we need to patch the PVC,
+     * however it's not needed -> so we skip unnecessary patches.
+     *
+     * @param current  Existing PVC
+     * @param desired  Desired PVC
+     */
+    protected void configureNormalizedStorageSizeIfEqual(PersistentVolumeClaim current, PersistentVolumeClaim desired) {
```

Comment:
Just thinking aloud ... isn't this a little of an hack? I mean ... maybe we should try to write the normalized value since the beginning (i.e. 1Ti instead of 1024Mi) instead of going through this check each time? Is it something you thought about? I would assume Kubernetes is rounding all the amounts which are exact multiple of Mi, Gi, and so on to the closer bigger unit. I would assume we are not the first one having this issue and maybe other operators do differently than this.

Reply:
Yes I thought about it, the issue what I was thinking was that you would do it for every reconciliation -> I mean the conversion of the values. Like I guess it would be the same thing, as this anyway does the conversion underneath and compares the values. No strong opinion on that TBH. I thought that Fabric8 provides it in the `Quantity` class, but just for the comparison stuff, there is nothing like convert method or something.

Reply:
The users should IMHO use the bigger units if possible (because anyway there will be confusion as we experienced in the thread I mentioned above) that why it's different unit when you set for example 1024Mi, why it is 1Gi.
But yeah this was one of the "simplest" solution without writing anything else.

Reply:
It's anyway for the ResourceDiff check which checks the JSON values. I wanted to skip it and just check the values instead of checking the JSON object, but that was complicated and I didn't find a simple solution to that.